NumberBaseball/NumberBaseball/main.swift

```diff
@@ -38,23 +41,80 @@ func startGame(numberOfChance: Int) {
     numbersByComputer = generateRandomNumbers()
     var numberOfChance = numberOfChance
     
-    while(numberOfChance > 0 && strikeCount < 3) {
-        numbersByUser = generateRandomNumbers()
-        let convertedNumbers = numbersByUser.map({ String($0) })
-        print("임의의 수 :", convertedNumbers.joined(separator: " "))
-        
+    while numberOfChance > 0 {
+        numbersByUser = getUserNumbers()
         updateStrikeAndBallCount()
         numberOfChance -= 1
-        
+        print("\(strikeCount) 스트라이크, \(ballCount) 볼")
         if strikeCount == 3 {
             print("사용자 승리!")
-        } else if numberOfChance == 0 {
-            print("컴퓨터 승리...!")
+            break
         }
-        
-        print("\(strikeCount) 스트라이크, \(ballCount) 볼")
         print("남은 기회 : \(numberOfChance)")
+        if numberOfChance == 0 {
+            print("컴퓨터 승리...!")
+        }
+    }
+}
+
+func printMenus() {
+    print("1. 게임시작")
+    print("2. 게임종료")
+    print("원하는 기능을 선택해주세요", terminator: " : ")
+}
+
+func printInputGuideline() {
+    print("숫자 3개를 띄어쓰기로 구분하여 입력해주세요.\n중복 숫자는 허용하지 않습니다.")
+    print("입력", terminator: " : ")
+}
+
+func playNumberBaseball(numberOfChance: Int) {
+    printMenus()
+    guard let userInput: String = readLine() else {
+        print("nil을 입력 하지마세요! 😡")
+        return }
+    switch userInput {
+    case "1":
+        startGame(numberOfChance: numberOfChance)
+        playNumberBaseball(numberOfChance: numberOfChance)
+    case "2":
+        return
+    default:
+        print("입력이 잘못되었습니다")
+        playNumberBaseball(numberOfChance: numberOfChance)
     }
 }
 
-startGame(numberOfChance: 9)
+func getUserNumbers() -> [Int] {
+    var userNumbers: [Int] = []
+    
+    while userNumbers == [] {
```

Comment:
헉... 발견 못했었네요 감사합니다.
여담으로  ```.count == 0```, ```.isEmpty()```, ```== []```  등 배열이 비었는지 확인하는 방법이 정말 다양한 것 같아요.

Reply:
isEmpty가 확실히 "비어있는지"라는 의미가 확 와닿네요..!

Reply:
맞습니다 ㅎㅎㅎ

[isEmpty 공식문서](https://developer.apple.com/documentation/swift/array/1688398-isempty)를 확인해보시면
만약 컬렉션 타입이 비어있는지 확인하려면, count 프로퍼티가 0 인지 비교하기 보다는 isEmpty 프로퍼티를 사용하도록 권장하고 있습니다. 😄 

> When you need to check whether your collection is empty, use the isEmpty property instead of checking that the count property is equal to zero.

주의하실 점은 isEmpty 는 메서드가 아니고 `연산 프로퍼티`입니다!
그래서 사용하실 때 괄호() 가 필요하지 않아요 ㅎㅎ